pkg/appgw/requestroutingrules.go

```diff
@@ -398,3 +399,21 @@ func (c *appGwConfigBuilder) mergePathMap(existingPathMap *n.ApplicationGatewayU
 
 	return existingPathMap
 }
+
+func printPathRule(pathRule n.ApplicationGatewayPathRule) string {
+	s := fmt.Sprintf("pathMapName=%s", *pathRule.Name)
+
+	if pathRule.BackendAddressPool != nil && pathRule.BackendAddressPool.ID != nil {
```

Comment:
would ID be allowed to be nil if BackendAddressPool is not nil?

Reply:
it shouldn't be - but checking for sanity